### ISSUE TITLE
IsExact does not handle `readonly` modifier

### ISSUE THREAD

**Issue by Yaojian** [dsherret/conditional-type-checks#20]
`IsExact` returns `true` when the only difference between two interfaces is the `readonly` modifier.

```ts
    interface A {
      value: string;
    }

    interface B {
      readonly value: string;
    }

    assert<IsExact<A, B>>(false);     // fails the compiler because `IsExact` returns `true`
```

I consider it as a bug.

**Comment by papb** [comment]
Agreed. I have been using [expect-type](https://github.com/mmkal/ts/tree/main/packages/expect-type) for this kind of thing, I believe it was inspired by this package, and has had a few bug fixes, and already covers correctly this case you reported.
